### PR TITLE
snap: introduce the StandardComponent type

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**
StandardComponent is the type of vanilla components that do not have any special behavior from the snapd perspective and that are simply mounted and made available to the snap.

### CODE REVIEW

**Review by zyga — APPROVED** [review]
Looks good to me.

Do we need to alter any documentation about the component type?

File: snap/types.go [hunk 1]
```diff
@@ -198,11 +198,13 @@ type ComponentType string
 const (
 	// TestComponent is just for testing purposes.
 	TestComponent ComponentType = "test"
+	// StandardComponent is for vanilla components with no special behavior.
+	StandardComponent ComponentType = "standard"
```
Comment (andrewphelpsj) [inline snap/types.go:202]: Maybe worth getting id of the `test` type, since they are functionally the same?

**Review by andrewphelpsj — APPROVED** [review]
Thanks!

**Review by pedronis — COMMENTED** [review]
thx, should we have change some of the spread tests using "test" type if have any to use this instead? or should we add a simple spread test for this?

**Comment by ernestl** [comment]
Blocked,  name needs to be Gustavo approved.

Awaiting response to: https://github.com/canonical/snapd/pull/14412#pullrequestreview-2299762444

**Comment by ernestl** [comment]
Changing milestone to 2.67. We did not manage to have meeting to confirm the naming time.

**Comment by alfonsosanchezbeato** [comment]
This new type has been approved and this can be merged now.

File: snap/types.go (hunk 1 above)
Comment (alfonsosanchezbeato) [reply snap/types.go:203]: We want to do that eventually, but not immediately as we have some snaps that we use in spread tests and have this type and that we uploaded to the store. We need to change those first.

**Review by alfonsosanchezbeato — COMMENTED** [review]
I have changed most tests now to use "standard" component type instead of "test".

**Review by pedronis — APPROVED** [review]
thank you

**Comment by ernestl** [comment]
Decided to cherry pick to 2.66

**Comment by ernestl** [comment]
Failures does not related to changes.

**Pull request merged**